src/pages/store/cart.tsx

```diff
@@ -0,0 +1,323 @@
+            />
+          </div>
+        )}
+
+        {/* Points Summary */}
+        <div className={`${styles.cartCard} ${styles.pointsCard}`}>
+          <Typography variant="h4/bold" component="h2" className={styles.header}>
+            {cartState !== CartState.CONFIRMED ? 'Membership Points' : 'Balance Details'}
+          </Typography>
+          <div className={styles.points}>
+            <div>
+              <Typography variant="h5/bold" component="p">
+                {cartState !== CartState.CONFIRMED ? 'Current' : 'Previous'} Balance
+              </Typography>
+              <Diamonds count={credits} />
+            </div>
+            <div>
+              <Typography variant="h5/bold" component="p">
+                Order Total
+              </Typography>
+              <div>
+                - <Diamonds count={orderTotal} />
+              </div>
+            </div>
+            <hr />
+            <div>
+              <Typography variant="h5/bold" component="p">
+                Remaining Balance
+              </Typography>
+              <Diamonds count={credits - orderTotal} />
+            </div>
+          </div>
+        </div>
+        <div className={styles.warning}>
+          {credits < orderTotal && (
+            <Typography variant="h5/regular" component="p">
+              You do not have enough membership points to checkout. Earn more membership points by
+              attending ACM events!
+            </Typography>
+          )}
+          {cartInvalid && (
+            <Typography variant="h5/regular" component="p">
+              Some items in your cart are not available. Remove them to check out!
+            </Typography>
+          )}
+        </div>
+      </div>
+    </div>
+  );
+};
+
+export default StoreCartPage;
+
+const getServerSidePropsFunc: GetServerSideProps = async ({ req, res }) => {
+  const AUTH_TOKEN = getServerCookie(CookieType.ACCESS_TOKEN, { req, res });
+
+  // recover saved items and reset cart to [] if it's invalid
+  let savedItems;
+  try {
+    const cartCookie = getServerCookie(CookieType.CART, { req, res });
+    savedItems = JSON.parse(cartCookie);
+    if (!Array.isArray(savedItems)) throw new Error();
+  } catch {
+    savedItems = [];
+  }
+
+  const savedCartPromises = Promise.all(
+    savedItems.map(async ({ itemUUID, optionUUID, quantity }: CookieCartItem) => {
+      try {
+        const { options, ...publicItem } = await getItem(AUTH_TOKEN, itemUUID);
+
+        // form the ClientCartItem from the PublicMerchItem
+        const ClientCartItem: Partial<ClientCartItem> = { ...publicItem, quantity };
+        ClientCartItem.option = options.find(
+          (option: PublicMerchItemOption) => option.uuid === optionUUID
+        );
+
+        // check for invalid cart cookie item
+        if (!ClientCartItem.option || typeof quantity !== 'number') throw new Error();
+        return ClientCartItem as ClientCartItem;
```

Comment:
I think you should make the variable name lowercase so this line doesn't look as confusing